### PR TITLE
Add eslint-plugin-ember-cleanup rules

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #patch# - backwards-compatible bug fix
 - [ ] #minor# - adding functionality in a backwards-compatible manner
 - [x] #major# - incompatible API change

# CHANGELOG

 * **Added** new rules from [`eslint-plugin-ember-cleanup`](https://github.com/onechiporenko/eslint-plugin-ember-cleanup)

### CODE REVIEW

**Comment by sglanzer** [comment]
Awesome!  LGTM - @dafortin feasibility of rolling this into the LTS in a short timeframe?

**Comment by sandersky** [comment]
:+1:

**Comment by dafortin** [comment]
@sglanzer It's doable but I will need an extra pair of hands and probably delay starting the new micro service of 1 week. I can start the new micro service after Dec. 9.

Note: We still have a lot of work to do in the next week:

File: README.md [hunk 1]
```diff
@@ -25,14 +25,62 @@ Enforces a `camelCase` variable naming convention. More details available [here]
...
+Checks for dependent keys with invalid `[]` usage
+
+#### `ember-cleanup/super-args`
+Check potential invalid calls `_super` without `...`
+
 ### `max-len`
 Enforces a maximum line length of 120 characters. More details available [here](http://eslint.org/docs/rules/max-len)
 
 ### `valid-jsdoc`
 Make sure your `JSDoc` comment blocks are valid/consistent. More details availalbe [here](http://eslint.org/docs/rules/valid-jsdoc).
```
Comment (juwara0) [inline README.md]: I noticed a typo. "More details availalbe".

**Comment by sglanzer** [comment]
@dafortin I can't (don't want to) delay the start of that micro-service work.  Let's see if any of the co-ops can be pulled to help on this and free you

**Comment by job13er** [comment]
Sorry, when I noticed that this failed build, I pretty much lost confidence in the plugin we were gonna bring in. That plus the pretty shoddy documentation (the rules aren't even organized alphabetically) is making me thing this may not be a good fit for us after all. We can re-open if someone wants to disagree.

**Comment by sglanzer** [comment]
All good, we can build equivalent rules on our own given some time

**Comment by notmessenger** [comment]
If this is ever revisited I want(ed) to leave a comment regarding the `ember-cleanup/no-typeof` plugin which proposes the use of `Ember.typeOf` instead of built-in `typeof` for some type checks.

https://github.com/softlayer/sl-ember-test-helpers/issues/100#issuecomment-124117617, which refers to https://github.com/emberjs/ember.js/issues/11673, identifies a bug in `Ember.typeOf` wherein Symbols are incorrectly identified as `object`.

**Pull request closed without merging**